app/Http/Middleware/CheckRole.php

```diff
@@ -3,27 +3,38 @@
 namespace Gladiator\Http\Middleware;
 
 use Closure;
-use Illuminate\Support\Facades\Auth;
+use Illuminate\Contracts\Auth\Guard;
 
 class CheckRole
 {
     /**
+     * The Guard implementation.
+     */
+    protected $auth;
+
+    /**
+     * Create a new filter instance.
+     * @param Guard $auth
+     */
+    public function __construct(Guard $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    /**
      * Handle an incoming request.
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
      * @return mixed
      */
-    public function handle($request, Closure $next, $guard = null)
+    public function handle($request, Closure $next, ...$roles)
     {
-        $roles = array_slice(func_get_args(), 2);
-
-        if (Auth::guest() || ! Auth::user()->hasRole($roles)) {
+        if ($this->auth->guest() || ! $this->auth->user()->hasRole($roles)) {
             if ($request->ajax()) {
                 return response('Unauthorized.', 401);
             } else {
-                // @TODO: respond with a more custom unauthorized access page or whatevs.
-                return redirect('/');
+                return view('/');
```

Comment:
I don't think you can use the `view()` function with a route, you would use it to serve a named view. So if you wanted to show the home page. you would do `view(pages.home)` 

But I think it might be fine to just leave the `return redirect('/')` instead.